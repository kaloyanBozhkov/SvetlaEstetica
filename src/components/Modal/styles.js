const styles = {
    modalOverlay: {
        position: 'fixed',
        backgroundVolor: 'rgba(0, 0, 0, 0.7)',
        zIndex: 'overlay',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
    },
    box: {
        position: 'fixed',
        zIndex: 'modal',
        backgroundColor: 'white',
        top: '50%',
        left: '50%',
        transform: 'translate(-50%, -50%)',
        padding: '4',
        maxWidth: { mdX: '90%' },
        minWidth: { mdX: '70%', base: '40rem', lg: '40rem' },
        overflow: { mdX: 'hidden' }
    },
    close: {
        position: 'absolute',
        top: '1.5rem',
        right: '2rem',
        color: 'main.800',
        fontWeight: 'bolder',
        fontSize: '1.4rem',
        cursor: 'pointer'
    },
    title: {
        paddingLeft: '0.5rem',
        borderLeft: '3px solid',
        borderColor: 'main.800',
        fontSize: '1.4rem'
      }
}

export default styles