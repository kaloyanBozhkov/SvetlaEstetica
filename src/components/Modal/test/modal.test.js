import React from 'react'
import { mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import { Modal } from '../Modal'
import { Provider } from 'react-redux'

const mockStore = configureStore()

describe('testing modal', () => {
  const store = mockStore({
  })

  beforeEach(() => {
    store.clearActions()
  })

  it('should render modal', () => {
    const modal = mount(
      <Provider store={store}>
        <Modal modal="someUnhandledModal" />
      </Provider>
    )
    expect(modal).toMatchSnapshot()
  })
  it('should fire close fn on cose icon click', () => {
    const fn = jest.fn(() => 1)
    const modal = mount(
      <Provider store={store}>
        <Modal modal="someUnhandledModal" onCloseModal={fn} />
      </Provider>
    )
    modal.find('[data-test-id="closeBtn"]').at(0).simulate('click')
    expect(fn).toHaveBeenCalled()
  })
  it('should render modal with defalt "modal is empty" msg if modal name provided is not handled', () => {
    const modal = mount(
      <Provider store={store}>
        <Modal modal="unhandledModalName" />
      </Provider>
    )

    expect(modal.find('[data-test-id="emptyMsg"]').at(0).text()).toEqual('Modal is empty, not configured!')
  })
  it('should return null if modal not specified', () => {
    const modal2 = mount(
      <Provider store={store}>
        <Modal modal={null} />
      </Provider>
    )
    expect(modal2).toEqual({})
  })
  it('should render cofirm password modal with expected contents and pass them the expected props', () => {
    const props = {
      text: 'hello world',
    }
    const modal = mount(
      <Provider store={store}>
        <Modal data={props} modal="confirm" />
      </Provider>
    )
    expect(modal).toMatchSnapshot()
  })

  it('should close modal when pressed esc', () => {
    const closeFn = jest.fn()

    // obj to keep track of eventlisteners
    const map = {}

    // replace addEventListener with jest fn
    document.addEventListener = jest.fn((event, cb) => {

      // keep ref to event
      map[event] = cb
    })

    const modal = mount(
      <Provider store={store}>
        <Modal modal="someModalName" onCloseModal={closeFn} />
      </Provider>
    )

    // simulate keypress
    map.keydown({ key: 'Escape' })

    expect(closeFn).toHaveBeenCalled()
  })
})
