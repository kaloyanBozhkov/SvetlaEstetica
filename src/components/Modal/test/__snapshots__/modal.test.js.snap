// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing modal should render cofirm password modal with expected contents and pass them the expected props 1`] = `
Array [
  <div
    className="[object Object] css-0"
  />,
  <div
    className="css-5grkxp"
  >
    <div
      className="css-qlehve"
      data-test-id="closeBtn"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-times fa-w-11 "
        data-icon="times"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 352 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div>
      <div
        className={
          Object {
            "margin": "5 -2 5 -2",
            "padding": "0 2",
          }
        }
      >
        <p />
      </div>
      <div
        className={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "height": "3rem",
            "margin": "-4",
            "marginTop": "2",
          }
        }
      />
    </div>
  </div>,
]
`;

exports[`testing modal should render modal 1`] = `
Array [
  <div
    className="[object Object] css-0"
  />,
  <div
    className="css-5grkxp"
  >
    <div
      className="css-qlehve"
      data-test-id="closeBtn"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-times fa-w-11 "
        data-icon="times"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 352 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <p
      className="css-18poma1"
      data-test-id="emptyMsg"
    >
      Modal is empty, not configured!
    </p>
  </div>,
]
`;
